src/pages/home/report/ReportActionsView.js

```diff
@@ -206,10 +206,11 @@ class ReportActionsView extends React.Component {
             this.openReportIfNecessary();
         }
 
-        // If the report is unread, we want to check if the number of actions has decreased. If so, then it seems that one of them was deleted. In this case, if the deleted action was the
-        // one marking the unread point, we need to recalculate which action should be the unread marker.
-        if (ReportUtils.isUnread(this.props.report) && ReportActionsUtils.filterReportActionsForDisplay(prevProps.reportActions).length > this.sortedAndFilteredReportActions.length) {
-            this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
+        // If the report action marking the unread point is deleted we need to recalculate which action should be the unread marker
+        if (this.state.newMarkerReportActionID && _.isEmpty(lodashGet(this.props.reportActions[this.state.newMarkerReportActionID], 'message[0].html'))) {
```

Comment:
NAB: At first I didn't understand why we need to check that `'message[0].html'` is empty. After debugging I see that when we delete a report action we only edit the message to be empty. It would be nice to have a comment explaining that.

Reply:
@neil-marcellini Indeed this is not clear and requires some debugging to find out, @thesahindia also asked the same in the above review. The PR is merged already please add a comment where necessary. Thanks for handling this quickly!

Making this conversation as resolved for now.